toxcore/group_chats.h

```diff
@@ -0,0 +1,859 @@
+ * Returns -1 if the message is too long.
+ * Returns -2 if the message pointer is NULL or length is zero.
+ * Returns -3 if the peer_id is invalid.
+ * Returns -4 if the message type is invalid.
+ * Returns -5 if the sender has the observer role.
+ * Returns -6 if the packet fails to send.
+ */
+int gc_send_private_message(const GC_Chat *chat, uint32_t peer_id, uint8_t type, const uint8_t *message,
+                            uint16_t length);
+
+/* Sends a custom packet to the group. If lossless is true, the packet will be lossless.
+ *
+ * Returns 0 on success.
+ * Returns -1 if the message is too long.
+ * Returns -2 if the message pointer is NULL or length is zero.
+ * Returns -3 if the sender has the observer role.
+ */
+int gc_send_custom_packet(const GC_Chat *chat, bool lossless, const uint8_t *data, uint32_t length);
+
+/* Toggles ignore for peer_id.
+ *
+ * Returns 0 on success.
+ * Returns -1 if the peer_id is invalid.
+ * Returns -2 if the caller attempted to ignore himself.
+ */
+int gc_toggle_ignore(GC_Chat *chat, uint32_t peer_id, bool ignore);
+
+/* Sets the group topic and broadcasts it to the group.
+ *
+ * Returns 0 on success.
+ * Returns -1 if the topic is too long.
+ * Returns -2 if the caller does not have the required permissions to set the topic.
+ * Returns -3 if the packet cannot be created or signing fails.
+ * Returns -4 if the packet fails
+ */
+int gc_set_topic(GC_Chat *chat, const uint8_t *topic, uint16_t length);
+
+/* Copies the group topic to topic. If topic is null this function has no effect.*/
+void gc_get_topic(const GC_Chat *chat, uint8_t *topic);
+
+/* Returns topic length. */
+uint16_t gc_get_topic_size(const GC_Chat *chat);
+
+/* Copies group name to groupname. If group_name is null this function has no effect. */
+void gc_get_group_name(const GC_Chat *chat, uint8_t *group_name);
+
+/* Returns group name length */
+uint16_t gc_get_group_name_size(const GC_Chat *chat);
+
+/* Copies the group password to password. If password is null this function has no effect. */
+void gc_get_password(const GC_Chat *chat, uint8_t *password);
+
+/* Returns the group password length. */
+uint16_t gc_get_password_size(const GC_Chat *chat);
+
+/* Returns group privacy state. */
+Group_Privacy_State gc_get_privacy_state(const GC_Chat *chat);
+
+/* Returns the group's topic lock state. */
+Group_Topic_Lock gc_get_topic_lock_state(const GC_Chat *chat);
+
+/* Returns the group peer limit. */
+uint32_t gc_get_max_peers(const GC_Chat *chat);
+
+/*
+ * Sets your own nick.
+ *
+ * Returns 0 on success.
+ * Returns -1 if group_number is invalid.
+ * Returns -2 if the length is too long.
+ * Returns -3 if the length is zero or nick is a NULL pointer.
+ * Returns -4 if the nick is already taken.
+ * Returns -5 if the packet fails to send.
+ */
+int gc_set_self_nick(const Messenger *m, int group_number, const uint8_t *nick, uint16_t length);
+
+/* Copies your own nick to nick. If nick is null this function has no effect. */
```

Comment:
This comment doesn't quite match the comment on the implementation. The version on the impl has better formatting and more information. That's probably true for a lot of others as well, this is just the first and only one I checked it for.